### ISSUE TITLE
How to match do/end pairs with %?

### ISSUE THREAD

**Issue by carlosgrell** [AstroNvim/AstroNvim#642]
Hey there, I tried matching do/end pairs in ruby with %, but it does not seem to work.

I tried out some configs and even managed to get Endwise to work when inserting `do` – though only `do`, not `def`. But when jumping to matching end with %, it never works.

I also tried adding and removing null-ls linters and formatters back and forth, but still no result.

Thanks in advance.

**Comment by mehalter** [comment]
I would check out this plugin and enable the treesitter Integration: https://github.com/andymass/vim-matchup

**Comment by jayp0521** [comment]
any reason this is not part of astronvim?

> AstroNvim is an aesthetic and feature-rich neovim config that is extensible and easy to use with a great set of plugins

*feature-rich* and *great set of plugins* would be fulfilled by doing this :)

**Comment by mehalter** [comment]
Yeah, I will look into this. The main issue is if a lot of people don't use this feature in particular it is a very heavy plugin and is all vimscript. On my laptop which is very modern hardware it adds like 13ms to the startup time so it's really slow :/

**Comment by jayp0521** [comment]
> Yeah, I will look into this. The main issue is if a lot of people don't use this feature in particular it is a very heavy plugin and is all vimscript. On my laptop which is very modern hardware it adds like 13ms to the startup time so it's really slow :/

you can defer loading until treesitter loads? I guess that is only if astronvim decides to only enable treesitter support by default. I don't think there is an issue with that as it comes with treesitter anyways. maybe the plugin can be opt in instead of opt out

**Comment by mehalter** [comment]
I'm not sure there's a super clean way to defer the plugin loading until treesitter actually loads a parser. As it is right now it is kinda opt in where you just need to add it to the user configuration. I can add a recipe to the documentation. We have discussed adding a section of like "recommended plugins" along with some basic recommended configuration for them. This is a really good example of what could go there with some easy copy/paste instructions for users :)

**Comment by carlosgrell** [comment]
I tried adding a vim.init file with 

```
Plug 'nvim-treesitter/nvim-treesitter'
lua <<EOF
require'nvim-treesitter.configs'.setup {
  matchup = {
    enable = true,              -- mandatory, false will disable the whole extension
    disable = { "c", "ruby" },  -- optional, list of language that will be disabled
    -- [options]
  },
}
EOF
```

But it didn't work because it doesn't work along with init.lua

**Comment by mehalter** [comment]
@carlosgrell here is a minimal user configuration file with the appropriate sections (this is a complete `user/init.lua` file)

```lua
return {
  plugins = {
    init = {
      { 'andymass/vim-matchup' },
    }
    treesitter = {
      matchup = {
        enable = true,
      },
    },
  },
}
```

**Comment by carlosgrell** [comment]
> @carlosgrell here is a minimal user configuration file with the appropriate sections (this is a complete `user/init.lua` file)
> 
> ```lua
> return {
>   plugins = {
>     init = {
>       { 'andymass/vim-matchup' },
>     },
>     treesitter = {
>       matchup = {
>         enable = true,
>       },
>     },
>   },
> }
> ```

Thanks!! It works now.

**Comment by jayp0521** [comment]
> I'm not sure there's a super clean way to defer the plugin loading until treesitter actually loads a parser. As it is right now it is kinda opt in where you just need to add it to the user configuration. I can add a recipe to the documentation. We have discussed adding a section of like "recommended plugins" along with some basic recommended configuration for them. This is a really good example of what could go there with some easy copy/paste instructions for users :)

It is up to you, but I have noticed documentation tends to get out of date in general vs code that is disabled as you are forced to update it

**Comment by mehalter** [comment]
I have evaluated this and I don't think there is a good way to load this plugin without doing it on start up along side `treesitter`. So I think for now we are going to leave it not in the provided plugins since this aligns with the contribution guidelines and goals of AstroNvim moving forward. It is also pretty easy to include the plugin in the user configuration. I am going to put together a new page in the docs with recommended plugins soon and add this in there with example setup for users that want it.
